presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java

```diff
@@ -0,0 +1,120 @@
+        this.checksumConverter = requireNonNull(checksumConverter, "checksumConverter is null");
+    }
+
+    protected abstract Statement getChecksumQuery(QueryObjectBundle queryBundle);
+
+    protected abstract boolean match(S controlObject, S testObject, QueryObjectBundle control, QueryObjectBundle test);
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected DdlMatchResult verify(
+            QueryObjectBundle control,
+            QueryObjectBundle test,
+            Optional<QueryResult<Void>> controlQueryResult,
+            Optional<QueryResult<Void>> testQueryResult,
+            ChecksumQueryContext controlContext,
+            ChecksumQueryContext testContext)
+    {
+        Statement controlChecksumQuery = getChecksumQuery(control);
+        Statement testChecksumQuery = getChecksumQuery(test);
+
+        controlContext.setChecksumQuery(formatSql(controlChecksumQuery));
+        testContext.setChecksumQuery(formatSql(testChecksumQuery));
+
+        String controlChecksum = getOnlyElement(callAndConsume(
+                () -> getHelperAction().execute(controlChecksumQuery, CONTROL_CHECKSUM, checksumConverter),
+                stats -> stats.getQueryStats().map(QueryStats::getQueryId).ifPresent(controlContext::setChecksumQueryId)).getResults());
+        String testChecksum = getOnlyElement(callAndConsume(
+                () -> getHelperAction().execute(testChecksumQuery, TEST_CHECKSUM, checksumConverter),
+                stats -> stats.getQueryStats().map(QueryStats::getQueryId).ifPresent(testContext::setChecksumQueryId)).getResults());
+
+        S controlObject;
+        S testObject;
+
+        try {
+            controlObject = (S) sqlParser.createStatement(controlChecksum, PARSING_OPTIONS);
+        }
+        catch (ParsingException e) {
+            return new DdlMatchResult(CONTROL_NOT_PARSABLE, Optional.of(e), controlChecksum, testChecksum);
+        }
+
+        try {
+            testObject = (S) sqlParser.createStatement(testChecksum, PARSING_OPTIONS);
+        }
+        catch (ParsingException e) {
+            return new DdlMatchResult(TEST_NOT_PARSABLE, Optional.of(e), controlChecksum, testChecksum);
+        }
+
+        return new DdlMatchResult(
+                match(controlObject, testObject, control, test) ? MATCH : MISMATCH,
+                Optional.empty(),
+                controlChecksum,
+                testChecksum);
+    }
+
+    @Override
+    protected DeterminismAnalysisDetails analyzeDeterminism(QueryObjectBundle control, DdlMatchResult matchResult)
+    {
+        throw new UnsupportedOperationException("analyzeDeterminism is not supported for DdlVerification");
+    }
+
+    @Override
+    protected Optional<String> resolveFailure(Optional<QueryObjectBundle> control, Optional<QueryObjectBundle> test, QueryContext controlQueryContext, Optional<DdlMatchResult> matchResult, Optional<Throwable> throwable)
+    {
+        return Optional.empty();
+    }
```

Comment:
Do we not want a ```failureResolveManager``` like DataVerification?

Reply:
Failure resolver is used to automatically resolve verification failures due to (test / checksum) query failures. For ddl verification, we can start with no auto resolution, and see if we actually see any query failures are actually false alerts.